File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const complimentNum = {}
        for (let i=0; i< nums.length; i ++){
            const reqNum = target - nums[i]
                console.log(complimentNum, reqNum)
            if (reqNum in complimentNum) {
                return [complimentNum[reqNum], i]
            }else {
                complimentNum[nums[i]] = i
            }
        }
        return []
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int lp = 1;
        int[] result = new int[nums.length];
        for(int i =0; i< nums.length; i++) {
            result[i] = lp;
            lp = lp * nums[i];
        }
        int rp = 1;
         for(int i =nums.length -1; i>=0 ; i--) {
            result[i] = result[i] * rp;
            rp = rp * nums[i];
        }
        return result;
    }
}  
